persistence/provider/sql/mysql/offsetstore.go

```diff
@@ -0,0 +1,81 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/dogmatiq/infix/eventstream"
+	"github.com/dogmatiq/infix/internal/x/sqlx"
+)
+
+// LoadOffset loads the last offset associated with the given source
+// application key sk. ak is the 'owner' application key.
+//
+// If there is no offset associated with the given source application key,
+// the offset is returned as zero and error as nil.
+func (driver) LoadOffset(
+	ctx context.Context,
+	db *sql.DB,
+	ak, sk string,
+) (eventstream.Offset, error) {
+	var o eventstream.Offset
+
```

Comment:
I've moved this variable declaration down to the scan. I'm invoking the "declare variables close to their usage" adage, here. I've made the same suggestion in the other SQL packages too.

```suggestion
```